### DISCUSSION TITLE
Syntax highlighting of code in `\directlua` argument

### DISCUSSION THREAD

**Discussion by leingang** [James-Yu/LaTeX-Workshop#4224] [Q&A] [answered]
I noticed that when I have lua code inside a `luacode` or `luacode*` environment, then the lua code is highlighted for lua syntax instead of LaTeX syntax. I love it!

But I have some use cases where I cannot use these environments, because I want to pass values as macros to the enclosing group. Since [an environment is a group](https://tex.stackexchange.com/a/464470/1402), any macro assignments disappear after `\end{luacode}`.

Is there a way to have the lua syntax highlighting inside the arguments to the non-environment markers like `\directlua` and `\luadirect`?

**Comment by jlelong** [comment, marked as answer]
Currently, it is not possible but it can certainly be added.
https://github.com/jlelong/vscode-latex-basics/issues/82

**Reply by leingang** [reply]
Thanks! I'll look forward to it.

**Reply by jlelong** [reply]
It has been implemented.

**Comment by leingang** [comment]
Wonderful, thanks!
